app/views/collection_numbers/_form.html.erb

```diff
@@ -1,9 +1,12 @@
-<% # form_tag(add_query_param(action: :new, id: @observation.id)) do %>
-<%= form_with model: :collection_number do |f| %>
+<% id = @observation&.id
+   id_path_segment = id ? "/#{id}" : nil
+   url = "#{collection_numbers_path}#{id_path_segment}" +
+         "?back=#{@back}&q=#{get_query_param}" %>
 
+<%= form_with model: @collection_number, url: url do |f| %>
   <%= f.submit button.t, class: "btn mx-auto mt-3" %>
 
-  <%= f.hidden_field :id, value: @observation.id %>
+  <%= f.hidden_field :id, value: id %>
```

Comment:
In the new syntax, this line throws an error if `@observation` is `nil`:
```ruby
ERROR["test_edit", #<Minitest::Reporters::Suite:0x000055dd37bb55c0 @name="CollectionNumbersControllerTest">, 11.624770464986796]
 test_edit#CollectionNumbersControllerTest (11.63s)
Minitest::UnexpectedError:         ActionView::Template::Error: no implicit conversion of nil into String
            app/views/collection_numbers/_form.html.erb:3:in `+'
            app/views/collection_numbers/_form.html.erb:3:in `_app_views_collection_numbers__form_html_erb__1435477874758573121_47204341240040'
            app/views/collection_numbers/edit.html.erb:18:in `_app_views_collection_numbers_edit_html_erb__3309629008407258624_47204347943700'
            app/controllers/application_controller.rb:226:in `catch_errors_and_log_request_stats'
            test/functional_test_case.rb:26:in `get'
            test/controller_extensions.rb:113:in `get_with_dump'
            test/controllers/collection_numbers_controller_test.rb:249:in `test_edit'
```

Reply:
Would it work to just make adding the hidden field conditional on @observation?  The old code does suggest that this page never actually gets called without @observation being set.

Reply:
Maybe the test can set `@observation`

Reply:
Thank you @mo-nathan & @nimmolo for this discussion and all the other comments above.
1. You've made the penny finally drop for me about how the forms work.
2. The problem in this discussion is fixed in Commit #196f99ca. There were two underlying causes: 
- form failing to set `:id` to `@observation.id`; 
- `edit` action failing to set `@observation`.